src/DiffOp/Diff/Diff.php

```diff
@@ -366,11 +356,31 @@ public function toArray( $valueConverter = null ) {
 	}
 
 	/**
+	 * @since 2.0
+	 *
+	 * @param mixed $target
+	 *
+	 * @return bool
+	 */
+	public function equals( $target ) {
```

Comment:
I'm afraid this is what PHP does when [comparing two ArrayObjects](https://phabricator.wikimedia.org/T92487#1113096).